drivers/hmis/app/models/hmis/client_alert.rb

```diff
@@ -0,0 +1,10 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::ClientAlert < Hmis::HmisBase
+  belongs_to :created_by, class_name: 'Hmis::User'
```

Comment:
It does apply, but I think it's OK. The database column is non-nullable, but there is no foreign key constraint, so its possible that the user could have been deleted. In that case, the `alert.created_by` would be null.

That's fine, generally, we tend not to add foreign keys for lots of things.. I think the main reasoning is that we don't want to have to cascade deletion. it's ok to keep these records around if the user gets deleted for whatever reason (which is not expected btw).

The only gotcha is that on the ClientAlert GraphQL type, the createdBy user field should be nullable. That way, the frontend can prepare for that scenario, rather than crash and burn (which is what GraphQL does if it tries to resolve a null value for a non-nullable schema field).